### PR TITLE
GeoUsage updates from ES parsing

### PR DESCRIPTION
**Pull request by kngai**
- Parsing fix for kvp
- Fix to error checks and handling
- Improved logger messages
- Parse the full user_agent details as the 12th token to end of log line
- Added IP number (derived from IP address) as part of OWSLogRecord class
- Added `crs`, `format`, `ows_resource` and `styles` as part of OWSLogRecord class
- `ows_resource` parsing compiled as one for the keys: `layer`, `layers`, `typename` and `coverageid`
- `crs` parsing compiled as one of the keys: `crs` and `subsettingcrs`
- `format` parsing compiled as one of the keys: `format` and `outputformat`
- Analyzer class fix for using non-WMSLogRecord or WMSLogRecord class type
- Added top user agents to GeoUsage log analyze command
- lower case name functions
- added test_dot2longip
- run_tests.py to elapsed times for slow unit tests after a set SLOW_TEST_THRESHOLD
- run_tests.py fix to open/read access_log and correct unit test cases

### CODE REVIEW

File: GeoUsage/log.py [hunk 1]
```diff
@@ -336,30 +409,52 @@ def test_time(intime, times, datetype='date'):
...
-        else:  # datetime.date comparison
+        else:   # datetime.date comparison
             LOGGER.debug('comparing date range')
             result = (intime.date() <= times[1].date() and
                       intime.date() >= times[0].date())
 
     return result
 
 
+def dot2LongIP(ip):
```
Comment (tomkralidis) [inline GeoUsage/log.py]: keep function names lowercase? `def dot2longip(ip):`

File: GeoUsage/tests/run_tests.py [hunk 2]
```diff
@@ -39,6 +42,26 @@
...
 THISDIR = os.path.dirname(os.path.realpath(__file__))
 
+SLOW_TEST_THRESHOLD = 0.3
+
+
+class TimeLoggingTestResult(TextTestResult):
+    """Outputs timing of slow running tests. Slowness threshold is the number
+    of seconds defined by SLOW_TEST_THRESHOLD"""
+
+    def startTest(self, test):
```
Comment (tomkralidis) [inline GeoUsage/tests/run_tests.py]: `def start_test(self, test):`

File: GeoUsage/tests/run_tests.py [hunk 3]
```diff
@@ -39,6 +42,26 @@
...
+        self._started_at = time.time()
+        super().startTest(test)
+
+    def addSuccess(self, test):
```
Comment (tomkralidis) [inline GeoUsage/tests/run_tests.py]: `def add_success(self, test):`

File: GeoUsage/tests/run_tests.py (hunk 2 above)
Comment (tomkralidis) [reply GeoUsage/tests/run_tests.py]: or do `startTest` and `addSuccess` override specific methods in `TextTestResult` ?

**Review by tomkralidis — CHANGES_REQUESTED** [review]

**Comment by kngai** [comment]
Changes requested have been pushed out (forced)

**Pull request merged**